src/main/kotlin/no/nav/syfo/narmestelederapi/NarmesteLederApi.kt

```diff
@@ -13,38 +15,42 @@ val log: Logger = LoggerFactory.getLogger("no.nav.syfo.syfonarmesteleder")
 
 fun Route.registrerNarmesteLederApi(narmesteLederClient: NarmesteLederClient) {
     get("/syfonarmesteleder/narmesteLeder/{narmesteLederAktorId}") {
-        withTraceInterceptor {
-            try {
-                val narmesteLederAktorId: String = call.parameters["narmesteLederAktorId"]?.takeIf { it.isNotEmpty() }
-                        ?: throw IllegalArgumentException("NarmesteLederAktorId mangler")
+        withContext(CoroutineMDCContext()) {
+            withTraceInterceptor {
```

Comment:
Er det mulig å slå sammen disse to? Gir det mening? Det blir litt sånn magisk at du må wrappe endepunktet ditt i begge to, ellers så feiler det..

Reply:
Det gir egentlig ikke mening. `CoroutineMDCContext` håndterer MDC-context for coroutines i alle situasjoner, mens `withTraceInterceptor` er knyttet spesifikt til REST-requesten.
Så:
- `CoroutineMDCContext` passer på at MDC-contexten alltid er lik i kjeden av coroutines
- `withTraceInterceptor` setter `Nav-Callid`++ i MDC-contexten i starten av ett REST-kall